index.html

```diff
@@ -101,19 +199,623 @@ <h3>
+              <li>Assert: |signal| is not [=AbortSignal/aborted=].
+              <li>
+                If |chunk| cannot be [=converted to an IDL value=] of type
+                {{BufferSource}}, reject |promise| with a {{TypeError}} and return
+                |promise|. Otherwise, save the result of the conversion in
+                |source:BufferSource|.
+              <li>
+                <li>
+                  [=In parallel=], run the following steps:
+                  <ol>
+                    <li>
+                      Invoke the operating system to write |bytes| to the socket.
+
+                      <div class="note">
+                        The operating system may return from this operation once
+                        |bytes| has been queued for transmission rather than after it
+                        has been transmitted.
+                      </div>
+                    <li>
+                      [=Queue a global task=] on the [=relevant global object=] of
+                      [=this=] using the [=TCP socket task source=] to run the
+                      following steps:
+                      <ol>
+                        <li>
+                          If the chunk was successfully written, [=resolve=] |promise| with `undefined`.
+
+                          <div class="note">
+                            [[STREAMS]] specifies that |writeAlgorithm| will only be
+                            invoked after the {{Promise}} returned by a previous
+                            invocation of this algorithm has resolved.
+                          </div>
+                        <li>
+                          If an operating system error was encountered, [=reject=]
+                          |promise| with an "{{NetworkError}}" {{DOMException}}.
+                        <li>
+                          If |the socket was disconnected|, run the following steps:
+                          <ol>
+                            <li>
+                              [=Reject=] |promise| with a "{{NetworkError}}"
+                              {{DOMException}}.
+                            <li>
+                              Invoke the steps to [=handle closing the writable
+                              stream=].
+                          </ol>
+                        <li>
+                          If |signal| is [=AbortSignal/aborted=], [=reject=] |promise|
+                          with |signal|'s [=AbortSignal/abort reason=].
+                      </ol>
+                  </ol>
+                <li>Return |promise|.
+            </ol>
+          <li>
+            Let |abortAlgorithm| be the following steps:
+            <ol>
+              <li>Let |promise| be [=a new promise=].
+              <li>Run the following steps [=in parallel=]:
+                <ol>
+                  <li>
+                    Invoke the operating system to shutdown the socket for writing.
+                  <li>
```

Comment:
You need to queue a task back on the TCP socket task source before these steps.

Reply:
Fixed!